### PR TITLE
kubelet: avoid printing sensitive containers informations

### LINKED ISSUE #122101: The kubelet may print sensitive container information if the container environment variables contain encrypted sensitive information.
### What happened?

When I set the log level of kubelet to 4.And the container environment variable configuration contains encrypted sensitive information.After starting the container, The kubelet prints all sensitive environment variable information.

In kubernetes 1.25，the code is located in method SyncPod of file kuberuntime_manager.go.
`klog.V(4).InfoS("Creating container in pod", "containerType", typeName, "container", spec.container, "pod", klog.KObj(pod))`

### What did you expect to happen?

Environment variable information should not be printed in kubelet logs because it may contain sensitive data.

### How can we reproduce it (as minimally and precisely as possible)?

Set the log level of kubelet to 4. Add sensitive information to the environment variables of the container. Start the container. Searching for 'Creating container in pod' in the log file of kubelet.

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by tmilox**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

If applied this commit will prevent printing sensitive containers informations in kubelet logs.

#### Which issue(s) this PR fixes:

Fixes #122101

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/kuberuntime/kuberuntime_manager.go [hunk 1]
```diff
@@ -1546,3 +1547,11 @@ func (m *kubeGenericRuntimeManager) ListMetricDescriptors(ctx context.Context) (
 func (m *kubeGenericRuntimeManager) ListPodSandboxMetrics(ctx context.Context) ([]*runtimeapi.PodSandboxMetrics, error) {
 	return m.runtimeService.ListPodSandboxMetrics(ctx)
 }
+
+func removeSensitiveContainerInformation(pod *v1.Pod) *v1.Pod {
```
Comment (kannon92) [inline pkg/kubelet/kuberuntime/kuberuntime_manager.go:1551]: Nit: maybe we should move this to helper.go and add a test for it?

otherwise, LGTM.

Comment (TommyStarK) [reply pkg/kubelet/kuberuntime/kuberuntime_manager.go:1551]: Sure thing, will do asap

Comment (TommyStarK) [reply pkg/kubelet/kuberuntime/kuberuntime_manager.go]: done in 9b42e548639

**Comment by VirrageS** [comment]
What about other logs where we print Pod or Container data? Why are you changing only this particular one?

Also regardless of verbosity you are going to make a copy of the Pod just to remove data. This doesn't sound good from performance perspective.

**Comment by TommyStarK** [comment]
I realised that I misunderstood the code, the container data are not printed where I made the changes. The `KObj`  function only returns an `ObjectRef` containing the name and namespace of the object. I am going to go ahead and close the PR to open a new one with a correct (and more efficient) proposal.

/close

**Pull request closed without merging**